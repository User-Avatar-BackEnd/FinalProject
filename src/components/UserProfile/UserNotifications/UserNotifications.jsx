import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { LoopCircleLoading } from 'react-loadingg';
import { useToasts } from 'react-toast-notifications';
import {getNotifications, notificationResponse} from '../../../store/ducks/user/user';
import selector from './UserNotifications.selector';

import styles from './UserNotifications.module.scss';
import UserIcon from '../../UserIcon/UserIcon';

const UserNotifications = () => {
  const { notifications, loading } = useSelector(selector)
  const dispatch = useDispatch();
  const { addToast } = useToasts();

  const token = localStorage.getItem('AUTH_TOKEN')

  useEffect( () => {
    dispatch(getNotifications(token))
  }, [])

  const acceptInvite = (e) => {
    dispatch(notificationResponse(1, e.target.getAttribute('data-id'), token, addToast))
  }

  const declineInvite = (e) => {
    dispatch(notificationResponse(-1, e.target.getAttribute('data-id'), token, addToast))
  }

  return (
    <div className={styles.UserNotifications}>
      {loading
        ? <LoopCircleLoading color={'orange'} style={{position: 'relative', margin: '200px auto'}}/>
        : <>
          {notifications.length === 0
            ? <div className={styles.placeholder}><h3>You have no notifications yet...</h3></div>
            : ''
          }
          {notifications.map(item => {
            return <div className={styles.notification} key={item.id}>
              <UserIcon type={'small'} rank={item.inviter?.rank}/>
              <span><b>{item.inviter?.login}</b> invited you to the board <b>{item.board.title}</b></span>
              <div className={styles.controlButtons}>
                <div className={styles.accept} data-id={item.id} onClick={acceptInvite}>
                  Accept
                </div>
                <div className={styles.decline} data-id={item.id} onClick={declineInvite}>
                  Decline
                </div>
              </div>
            </div>
          })}
          </>
      }
    </div>
  );

}

export default UserNotifications;