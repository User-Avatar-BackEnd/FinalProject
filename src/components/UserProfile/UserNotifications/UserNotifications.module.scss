.UserNotifications {
  width: 100%;
  height: 100vh;
  overflow-y: scroll;

  & .notification {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    width: 50%;
    min-width: 300px;
    height: 150px;
    background-color: #f3f3f3;
    margin: 30px auto;
    border-radius: 10px;
    font-size: 24px;
    padding: 10px;
    box-sizing: border-box;
    color: #000;
    box-shadow: rgba(100, 100, 111, 0.2) 0 7px 29px 0;

    & .controlButtons {
      display: flex;
      justify-content: space-between;
      font-size: 28px;
      width: 60%;
      margin: 0 auto;

      & .accept, & .decline {
        height: 40px;
        width: 120px;
        text-align: center;
        line-height: 40px;
        border-radius: 3px;
        color: #fff;
        cursor: pointer;
        transition: background-color .3s;
      }

      & .accept {
        background-color: seagreen;

        &:hover {
          background-color: darken(green, 5%);
        }
      }

      & .decline {
        background-color: indianred;

        &:hover {
          background-color: darken(red, 10%);
        }
      }
    }
  }
}