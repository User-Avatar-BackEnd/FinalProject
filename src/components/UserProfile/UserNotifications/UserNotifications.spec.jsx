import React from 'react';
import { shallow } from 'enzyme';

import UserNotifications from './UserNotifications';

describe('UserNotifications component', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<UserNotifications />);
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it('should match snapshot', () => {

    expect(wrapper).toMatchSnapshot();
  });
});