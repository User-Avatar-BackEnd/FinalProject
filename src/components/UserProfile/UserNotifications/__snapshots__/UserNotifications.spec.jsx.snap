// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminEvents component should match snapshot 1`] = `
<div
  className="AdminEvents"
/>
`;
